sdk/storage/azure-storage-common/inc/azure/storage/common/reliable_stream.hpp

```diff
@@ -48,17 +50,17 @@ namespace Azure { namespace Storage {
     // Configuration for the re-triable stream
     ReliableStreamOptions const m_options;
     // callback to get a bodyStream in case Read operation fails
-    HTTPGetter m_httpGetter;
+    internal_::HttpGetter m_httpGetter;
     // Options to use when getting a new bodyStream like current offset
-    HttpGetterInfo m_retryInfo;
+    internal_::HttpGetterInfo m_retryInfo;
 
     int64_t OnRead(uint8_t* buffer, int64_t count, Azure::Core::Context const& context) override;
 
   public:
     explicit ReliableStream(
         std::unique_ptr<Azure::Core::IO::BodyStream> inner,
         ReliableStreamOptions const options,
-        HTTPGetter httpGetter)
+        internal_::HttpGetter httpGetter)
```

Comment:
That isn't sufficient because the `ReliableStream` is a public type and it accepts `HttpGetter` as a parameter to its ctor.

We shouldn't be exposing `_internal` types like this.

Reply:
Then we should keep these types in public namespace. Just rename `HTTPGetter` to `HttpGetter`

Reply:
Can we re-write some of the calling code to not use `HttpGetterInfo` or the `std::function` callback?

Reply:
From offline discussion, make `ReliableBodyStream` internal and remove the `HttpGetter` and `HttpGetterInfo` types. The names of those types didn't really match the intended purpose anyway.